pkg/manifests/manifests.go

```diff
@@ -1358,9 +1362,156 @@ func (f *Factory) PrometheusK8s(host string, grpcTLS *v1.Secret, trustedCABundle
+			})
+		}
+
+		auth := yaml2.MapSlice{}
+		if alertmanagerConfig.BearerToken != nil {
+			if alertmanagerConfig.BearerToken.Name == "" {
+				return nil, errors.Errorf("bearer token secret name not found")
+			}
+			if alertmanagerConfig.BearerToken.Key == "" {
+				return nil, errors.Errorf("bearer token secret key not found")
+			}
+			auth = append(auth, yaml2.MapItem{
+				Key: "credentials_file", Value: fmt.Sprintf("/etc/prometheus/secrets/%s/%s",
+					alertmanagerConfig.BearerToken.Name, alertmanagerConfig.BearerToken.Key),
+			})
+		}
+		if len(auth) != 0 {
+			cfg = append(cfg, yaml2.MapItem{
+				Key:   "authorization",
+				Value: auth,
+			})
+		}
+
+		tlsConfig := yaml2.MapSlice{}
+		if alertmanagerConfig.TLSConfig.CA != nil {
+			if alertmanagerConfig.TLSConfig.CA.Name == "" {
+				return nil, errors.Errorf("ca secret name not found")
+			}
+			if alertmanagerConfig.TLSConfig.CA.Key == "" {
+				return nil, errors.Errorf("ca secret key not found")
+			}
+			tlsConfig = append(tlsConfig, yaml2.MapItem{
+				Key: "ca_file", Value: fmt.Sprintf("/etc/prometheus/secrets/%s/%s",
+					alertmanagerConfig.TLSConfig.CA.Name, alertmanagerConfig.TLSConfig.CA.Key),
+			})
+		}
+		if alertmanagerConfig.TLSConfig.Cert != nil {
+			if alertmanagerConfig.TLSConfig.Cert.Name == "" {
+				return nil, errors.Errorf("cert secret name not found")
+			}
+			if alertmanagerConfig.TLSConfig.Cert.Key == "" {
+				return nil, errors.Errorf("cert secret key not found")
+			}
+			tlsConfig = append(tlsConfig, yaml2.MapItem{
+				Key: "cert_file", Value: fmt.Sprintf("/etc/prometheus/secrets/%s/%s",
+					alertmanagerConfig.TLSConfig.Cert.Name, alertmanagerConfig.TLSConfig.Cert.Key),
+			})
+		}
+		if alertmanagerConfig.TLSConfig.Key != nil {
+			if alertmanagerConfig.TLSConfig.Key.Name == "" {
+				return nil, errors.Errorf("key secret name not found")
+			}
+			if alertmanagerConfig.TLSConfig.Key.Key == "" {
+				return nil, errors.Errorf("key secret key not found")
+			}
+			tlsConfig = append(tlsConfig, yaml2.MapItem{
+				Key: "key_file", Value: fmt.Sprintf("/etc/prometheus/secrets/%s/%s",
+					alertmanagerConfig.TLSConfig.Key.Name, alertmanagerConfig.TLSConfig.Key.Key),
+			})
+		}
+		if alertmanagerConfig.TLSConfig.ServerName != "" {
+			tlsConfig = append(tlsConfig, yaml2.MapItem{
+				Key: "server_name", Value: alertmanagerConfig.TLSConfig.ServerName,
+			})
+		}
+		tlsConfig = append(tlsConfig, yaml2.MapSlice{
+			{Key: "insecure_skip_verify", Value: alertmanagerConfig.TLSConfig.InsecureSkipVerify},
+		}...)
+
+		cfg = append(cfg, yaml2.MapItem{
+			Key:   "tls_config",
+			Value: tlsConfig,
+		})
+
+		if len(alertmanagerConfig.StaticConfigs) > 0 {
+			sc := yaml2.MapSlice{
+				{Key: "targets", Value: alertmanagerConfig.StaticConfigs},
+			}
+			cfg = append(cfg, yaml2.MapItem{
+				Key:   "static_configs",
+				Value: []yaml2.MapSlice{sc},
+			})
+		}
+
+		cfgs = append(cfgs, cfg)
+	}
+
+	if len(cfgs) == 0 {
```

Comment:
can it happen since we already return early if `f.config.ClusterMonitoringConfiguration.PrometheusK8sConfig.AlertmanagerConfigs` is zero-length?

Reply:
Good catch. Yes. it can be removed now.